test/semantics/label14.f90

```diff
@@ -13,22 +13,121 @@
 ! limitations under the License.
 
 ! RUN: ${F18} -funparse-with-symbols %s 2>&1 | ${FileCheck} %s
-! CHECK: A DO loop should terminate with an END DO or CONTINUE inside its scope
-! CHECK: A DO loop should terminate with an END DO or CONTINUE inside its scope
-! CHECK: A DO loop should terminate with an END DO or CONTINUE inside its scope
 
+module iso_fortran_env
+  type :: team_type
+  end type
+end
+
+subroutine foo0()
   do 1 j=1,2
     if (.true.) then
+! CHECK: A DO loop should terminate with an END DO or CONTINUE
 1   end if
   do 2 k=1,2
     do i=3,4
       print*, i+k
+! CHECK: A DO loop should terminate with an END DO or CONTINUE
```

Comment:
I think this test will pass if the message comes out once anywhere in the source. It's not checking that the error is emitted for each place the CHECK comment appears.

It would be a better test if it were split into separate source files, each with one error. Or better yet, switch to using test_errors.sh.

Reply:
Thanks, it was indeed not testing as I intended.
I tried using `test_errors.sh` with `-Mstandard -Werror` but that did not work because `-Werror` does not add the `error:` prefix to warnings (which I think is correct, but annoying in this case).
So I went the other way you suggested and split the tests into different sources. To make it more valuable, I also added checks to test if the label-do were rewritten as non-label-do.